packages/xo-server-backup-reports/src/index.js

```diff
@@ -145,73 +220,176 @@ class BackupReportsXoPlugin {
     this._xo.on('job:terminated', this._report)
   }
 
-  test({ runId }) {
-    return this._backupNgListener(undefined, undefined, undefined, runId)
+  async test({ runId }) {
+    return this._report(runId)
   }
 
   unload() {
     this._xo.removeListener('job:terminated', this._report)
   }
 
-  _wrapper(status, job, schedule, runJobId) {
-    if (job.type === 'metadataBackup') {
-      return
-    }
+  async _report(runJobId, { type, status, force } = {}) {
+    const xo = this._xo
+    try {
+      if (type === 'call') {
+        return this._listener(status)
+      }
 
-    return new Promise(resolve =>
-      resolve(
-        job.type === 'backup'
-          ? this._backupNgListener(status, job, schedule, runJobId)
-          : this._listener(status, job, schedule, runJobId)
-      )
-    ).catch(logError)
+      const log = await xo.getBackupNgLogs(runJobId)
+      if (log === undefined) {
+        throw new Error(`no log found with runId=${JSON.stringify(runJobId)}`)
+      }
+
+      const reportWhen = log.data.reportWhen
+      if (
+        !force &&
+        (reportWhen === 'never' ||
+          (reportWhen === 'failure' && log.status === 'success'))
+      ) {
+        return
+      }
+
+      if (type === undefined) {
+        type = await xo.getJob(log.jobId).then(({ type }) => type)
+      }
+      return type === 'backup'
+        ? this._backupNgListener(log, runJobId, force)
+        : this._metadataBackupListener(log, runJobId, force)
```

Comment:
Why not passing the `job` and the `schedule` because they are used in both cases?